percona/controllers/perconapgbackup_controller.go

```diff
@@ -0,0 +1,190 @@
+	if err := r.Client.Get(ctx, request.NamespacedName, pgBackup); err != nil {
+		// NotFound cannot be fixed by requeuing so ignore it. During background
+		// deletion, we receive delete events from cluster's dependents after
+		// cluster is deleted.
+		if err = client.IgnoreNotFound(err); err != nil {
+			log.Error(err, "unable to fetch PerconaPGBackup")
+		}
+		return reconcile.Result{}, err
+	}
+
+	if pgBackup.Status.State == v2beta1.BackupSucceeded || pgBackup.Status.State == v2beta1.BackupFailed {
+		return reconcile.Result{}, nil
+	}
+
+	pgCluster := &v2beta1.PerconaPGCluster{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pgBackup.Spec.PGCluster, Namespace: request.Namespace}, pgCluster)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "get PostgresCluster")
+	}
+
+	switch pgBackup.Status.State {
+	case v2beta1.BackupNew:
+		if err := startBackup(ctx, r.Client, pgCluster, pgBackup); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "start backup")
+		}
+
+		pgBackup.Status.State = v2beta1.BackupStarting
+		if err := r.Client.Status().Update(ctx, pgBackup); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "update PGBackup status")
+		}
+
+		log.Info("Backup is starting")
+		return reconcile.Result{}, nil
+	case v2beta1.BackupStarting:
+		job, err := findBackupJob(ctx, r.Client, pgCluster, pgBackup)
+		if err != nil {
+			if errors.Is(err, ErrBackupJobNotFound) {
+				log.Info("Waiting for backup to start")
+
+				return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+			}
+			return reconcile.Result{}, errors.Wrap(err, "find backup job")
+		}
+
+		pgBackup.Status.State = v2beta1.BackupRunning
+		pgBackup.Status.JobName = job.Name
+
+		if err := r.Client.Status().Update(ctx, pgBackup); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "update PGBackup status")
+		}
+
+		return reconcile.Result{}, nil
+	case v2beta1.BackupRunning:
+		job := &batchv1.Job{}
+		err := r.Client.Get(ctx, types.NamespacedName{Name: pgBackup.Status.JobName, Namespace: pgBackup.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "get backup job")
+		}
+
+		if jobCompleted(job) {
+			pgBackup.Status.State = v2beta1.BackupSucceeded
+
+			if err := r.Client.Status().Update(ctx, pgBackup); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "update PGBackup status")
+			}
+
+			log.Info("Backup succeeded")
+
+			return reconcile.Result{}, nil
+		}
+
+		if jobFailed(job) {
+			pgBackup.Status.State = v2beta1.BackupFailed
+
+			if err := r.Client.Status().Update(ctx, pgBackup); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "update PGBackup status")
+			}
+
+			log.Info("Backup failed")
+
+			return reconcile.Result{}, nil
+		}
+
+		log.Info("Waiting for backup to complete")
+
+		return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+	case v2beta1.BackupSucceeded, v2beta1.BackupFailed:
+		return reconcile.Result{}, nil
```

Comment:
Redundant check
```suggestion
```

Reply:
we have exhaustiveness check in golangci-lint config. I added these to conform that. I converted it to default case tough.